test/AutoDiff/validation-test/address_sanitizer.swift

```diff
@@ -0,0 +1,24 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-build-swift %s -sanitize=address -o %t/sr13973
+// RUN: %target-run %t/sr13973
+
+// REQUIRES: executable_test
+// REQUIRES: asan_runtime
+
+import _Differentiation
+
+struct SR13973 {
+  let x: Float = 0
+
```

Comment:
Can I have a SIL test?

Reply:
For changes like this, both are important!

Reply:
I added a FileCheck to test/AutoDiff/SILOptimizer/differentiation_function_canonicalization.sil. It's pretty much the same thing that I do in this file, except in SIL.